src/fights.ts

```diff
@@ -323,157 +319,137 @@ function bestWitchessPiece() {
-        ) {
-          if (["Macrometeorite", "Powerful Glove"].includes(fightSource.name)) {
+      // PROFESSOR COPIES
+      if (have($familiar`Pocket Professor`)) {
+        const profCopies = [
+          {
+            property: "_garbo_meatChain",
+            maximizeParameters: [], // implicitly maximize against meat
+            macro: firstChainMacro(),
+            goalMaximize: (requirements: Requirement) => meatOutfit(true, requirements),
+          },
+          {
+            property: "_garbo_weightChain",
+            maximizeParameters: ["Familiar Weight"],
+            macro: secondChainMacro(),
+            goalMaximize: (requirements: Requirement) =>
+              maximizeCached(requirements.maximizeParameters, requirements.maximizeOptions),
+          },
+        ];
+
+        for (const profCopy of profCopies) {
+          const { property, maximizeParameters, macro, goalMaximize } = profCopy;
+          const fightSource = getNextEmbezzlerFight();
+          if (!fightSource) return;
+          if (get(property, false)) continue;
+
+          if (fightSource.gregariousReplace) {
             saberCrateIfDesired();
           }
-          withMacro(firstChainMacro(), () =>
-            fightSource.run({
-              macro: firstChainMacro(),
-            })
+
+          useFamiliar($familiar`Pocket Professor`);
+          goalMaximize(
+            Requirement.merge([
+              new Requirement(maximizeParameters, {
+                forceEquip: $items`Pocket Professor memory chip`,
+              }),
+              ...fightSource.requirements,
+            ])
           );
-          embezzlerLog.initialEmbezzlersFought +=
-            1 + get("_pocketProfessorLectures") - startLectures;
-        }
-        set("_garbo_meatChain", true);
-        postCombatActions();
-      }
 
-      startWandererCounter();
-
-      // SECOND EMBEZZLER CHAIN
-      if (have($familiar`Pocket Professor`) && !get<boolean>("_garbo_weightChain", false)) {
-        const startLectures = get("_pocketProfessorLectures");
-        const fightSource = getNextEmbezzlerFight();
-        if (!fightSource) return;
-        useFamiliar($familiar`Pocket Professor`);
-        const requirements = Requirement.merge([
-          new Requirement(["Familiar Weight"], {
-            forceEquip: $items`Pocket Professor memory chip`,
-          }),
-          ...fightSource.requirements,
-        ]);
-        maximizeCached(requirements.maximizeParameters, requirements.maximizeOptions);
-        if (
-          get("_pocketProfessorLectures") <
-          2 + Math.ceil(Math.sqrt(familiarWeight(myFamiliar()) + weightAdjustment()))
-        ) {
-          if (["Macrometeorite", "Powerful Glove"].includes(fightSource.name)) {
-            saberCrateIfDesired();
-          }
-          withMacro(secondChainMacro(), () =>
+          if (get("_pocketProfessorLectures") < pocketProfessorCopies()) {
```

Comment:
Seeing it in action, I'd rename `pocketProfessorCopies` to `potentialPocketProfessorLectures` or `pocketProfectorLectureMax` or something